app/upgrades/v1_6.go

```diff
@@ -67,17 +76,163 @@ func V010600rc0UpgradeHandler(
+			HostEnabled: true,
+			AllowMessages: []string{
+				"/cosmos.bank.v1beta1.MsgSend",
+				"/cosmos.bank.v1beta1.MsgMultiSend",
+				"/quicksilver.interchainstaking.v1.MsgSignalIntent",
+				"/quicksilver.interchainstaking.v1.MsgRequestRedemption",
+				"/quicksilver.participationrewards.v1.MsgSubmitClaim",
+				"/cosmos.authz.v1beta1.MsgGrant",
+				"/cosmos.authz.v1beta1.MsgRevoke",
+				"/ibc.applications.transfer.v1.MsgTransfer",
+			},
+		})
+
+		channels := map[string]string{
+			"osmo-test-5":     "channel-39",
+			"provider":        "channel-0",
+			"elgafar-1":       "channel-1",
+			"regen-redwood-1": "channel-2",
+		}
+
+		ctx.Logger().Info("Set TransferChannel field for zones")
+		appKeepers.InterchainstakingKeeper.IterateZones(ctx, func(index int64, zone *icstypes.Zone) (stop bool) {
+			zone.TransferChannel = channels[zone.ChainId]
+			appKeepers.InterchainstakingKeeper.SetZone(ctx, zone)
+			return false
+		})
+
+		ctx.Logger().Info("Removing incorrect IBC denom for LiquidAllowedDenomProtocolData")
+		appKeepers.ParticipationRewardsKeeper.IteratePrefixedProtocolDatas(ctx, types.GetPrefixProtocolDataKey(types.ProtocolDataTypeLiquidToken), func(idx int64, _ []byte, data types.ProtocolData) bool {
+			pd, err := types.UnmarshalProtocolData(types.ProtocolDataTypeLiquidToken, data.Data)
+			if err != nil {
+				return false
+			}
+			token, _ := pd.(*types.LiquidAllowedDenomProtocolData)
+
+			if token.ChainID == ctx.ChainID() {
+				return false
+			}
+
+			channel, found := appKeepers.IBCKeeper.ChannelKeeper.GetChannel(ctx, "transfer", channels[token.ChainID])
+			if !found {
+				panic(fmt.Errorf("unable to find channel %s", channels[token.ChainID]))
+			}
+
+			// derive the correct ibc denom; if it does not match then remmove it.
+			correctIbc := utils.DeriveIbcDenom("transfer", channel.Counterparty.ChannelId, "transfer", channels[token.ChainID], token.QAssetDenom)
+			if token.IbcDenom != correctIbc {
+				ctx.Logger().Info(fmt.Sprintf("incorrect IBC denom %s for LiquidAllowedDenomProtocolData %s, expected %s. removing", token.IbcDenom, token.QAssetDenom, correctIbc))
+				appKeepers.ParticipationRewardsKeeper.DeleteProtocolData(ctx, pd.GenerateKey())
+			}
+			return false
+		})
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
 // =========== PRODUCTION UPGRADE HANDLER ===========
 
-func V010600UpgradeHandler(
+func V010601UpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
 	appKeepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
-		// no action yet.
+		ctx.Logger().Info("Updating setting block params; 2MB max_bytes, 150M max_gas")
+		appKeepers.ParamsKeeper.
+			Subspace(baseapp.Paramspace).
+			WithKeyTable(paramstypes.ConsensusParamsKeyTable()).
+			Set(ctx, baseapp.ParamStoreKeyBlockParams, abci.BlockParams{
+				MaxBytes: 2072576,
+				MaxGas:   150000000,
+			})
```

Comment:
I have added a test for v1.6.1 handler(#1612), and I got an error on this code
 `test panicked: subspace already occupied` since baseapp.Paramspace has already initialized [here](https://github.com/quicksilver-zone/quicksilver/blob/main/app/keepers/keepers.go#L207)